sklearn/manifold/tests/test_spectral_embedding.py

```diff
@@ -117,33 +126,64 @@ def test_spectral_embedding_two_components(seed=36):
     true_label[0:n_sample] = 1
 
     se_precomp = SpectralEmbedding(
-        n_components=1, affinity="precomputed", random_state=np.random.RandomState(seed)
+        n_components=1,
+        affinity="precomputed",
+        random_state=np.random.RandomState(seed),
+        eigen_solver=eigen_solver,
     )
-    embedded_coordinate = se_precomp.fit_transform(affinity)
-    # Some numpy versions are touchy with types
-    embedded_coordinate = se_precomp.fit_transform(affinity.astype(np.float32))
-    # thresholding on the first components using 0.
-    label_ = np.array(embedded_coordinate.ravel() < 0, dtype="float")
-    assert normalized_mutual_info_score(true_label, label_) == pytest.approx(1.0)
+    for dtype in [np.float32, np.float64]:
+        if eigen_solver == "amg" and not amg_loaded:
+            # with pytest.raises(
+            #     ValueError, match="The eigen_solver was set to 'amg', but"
+            # ):
+            #     embedded_coordinate = se_precomp.fit_transform(
+            #         affinity.astype(dtype)
+            #     )
```

Comment:
Why is this commented out?

Reply:
I was trying to check for the `pyamg not installed` error here, but this would not pass pytest, so there was something wrong in the way these lines were written. Could you please advice how to do it correctly?

Reply:
This appears to be the last issue to resolve

Reply:
Maybe this would help?

https://github.com/scikit-learn/scikit-learn/pull/21534/files#r742771584

In any case, feel free to let it fail on the CI rather than pushing the commented out version so that everybody can see the error message without having to pull the branch and run the test locally.

Reply:
Thanks, fixed, see https://github.com/scikit-learn/scikit-learn/pull/21534/files#r742915773